### DISCUSSION TITLE
Persistant storage solutions

### DISCUSSION THREAD

**Discussion by az-arteria** [qdrant/qdrant#4861] [General]
Hi,

I wanted to ask a somewhat open-ended question about deploying Qdrant and using storage solutions with it.

Let's say I have a Kubernetes cluster with Qdrant containers deployed in the cloud. What would you suggest to use as the persistent storage for these Qdrant containers? Ideally I would prefer to use something like an S3 bucket to store all the data collections. Is there a way to accomplish this in such an environment?

I know that managed cloud and hybrid cloud exist, but let's assume for this question that I would like to handle these things myself without using them.

**Comment by timvisee** [comment]
Qdrant needs block level storage. It is a database/search engine and so it needs sufficient control over what is on disk.

S3 does not satisfy this and can therefore not be used. Nor does NFS work. Please see the listed storage requirements [here](https://qdrant.tech/documentation/guides/installation/#storage).

Though S3 doesn't work for regular storage, we do support it for collection snapshots. Read more about this [here](https://qdrant.tech/documentation/concepts/snapshots/#storage).

You'd be correct that [Qdrant cloud](https://cloud.qdrant.io/) will arrange all of this for you.

**Reply by az-arteria** [reply]
Thanks for the info. So in this case, I'm assuming Amazon EBS is an acceptable storage choice considering that it's block storage. Do you have any examples or thoughts on using Qdrant with EBS?

**Reply by timvisee** [reply]
Yes, using EBS should be fine. I don't think we have any example for this at hand. Note that while EBS is fine, EFS is not.
